lib/app/features/core/views/pages/error_page.dart

```
@@ -20,30 +22,40 @@ class ErrorPage extends HookConsumerWidget {
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final showDebugInfo = ref.watch(envProvider.notifier).get<bool>(EnvVariable.SHOW_DEBUG_INFO);
-    final noInternetConnection = !ref.watch(hasInternetConnectionProvider);
+    final hasInternetConnection = ref.watch(hasInternetConnectionProvider);
+    final prevNoInternetConnection = usePrevious(!hasInternetConnection);
+    useEffect(
+      () {
+        if (prevNoInternetConnection.falseOrValue && hasInternetConnection) {
+          ref.invalidate(initAppProvider);
```

asking to be sure  - why do we invalidate it when has a connection?   i thought we just need to hide this error modal.